aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -307,16 +309,17 @@ private void assertRemoteSubscriptionsStarted() {
     }
 
     /**
-     * Check that there were no interactions between the SyncProcessor
-     * and the model provider. This is used to verify that the synchronization
-     * processes don't start if there's no API configured.
+     * Check that the SyncProcessor did not start by asserting that the
+     * only interaction with the API category was triggered by the getPlugins
+     * method.
+     * @param mockApi Mock or spy of the ApiCategory being used for the test.
      */
-    private void assertSyncProcessorNotStarted() {
-        boolean syncProcessorNotInvoked = mockingDetails(modelProvider)
-            .getInvocations()
-            .stream()
-            .noneMatch(invocation -> invocation.getLocation().getSourceFile().contains("SyncProcessor"));
-        assertTrue(syncProcessorNotInvoked);
+    private void assertSyncProcessorNotStarted(ApiCategory mockApi) {
```

Comment:
Can we do this by checking for an event on the hub, instead of by using mockito invocation verification?

Reply:
That will probably be a little trickier since we'd be checking for the absence of an event. It's probably a useful pattern to create in the `HubAccumulator`, but I'd prefer to do that as a separate PR if that's OK.